internal/pipeline/sign.go

```diff
@@ -111,7 +111,7 @@ func (s *Signature) Verify(env map[string]string, sf SignedFielder, keySet jwk.S
 	}
 
 	_, err = jws.Verify([]byte(s.Value),
-		jws.WithKeySet(keySet),
+		jws.WithKeySet(keySet, jws.WithRequireKid(false)),
```

Comment:
without this change, all keys would have to have key ids. while the [keygen utility](https://github.com/buildkite/agent/pull/2415) we wrote will always output keys with key ids, we don't enforce that people use our tooling, and should let users not use key ids if they don't want to.

Reply:
hmm, one downside of this change is that it makes the error messaging for "you don't have the right keys for verification" significantly worse. When we required key IDs, the message was something like
```
WARNING: Job verification failed: invalid signature: key provider 0 failed: failed to find key with key ID "my-cool-key" in key set
```

which tells you a fair bit about the problem.

with `jws.WithRequireKid(false)`, the error message is:
```
⚠️ WARNING: Job verification failed: invalid signature: could not verify message using any of the signatures or keys
```

which isn't too too much worse, but it's the same as if the job couldn't be verified (ie someone did crimes on the job).

i'm in two minds as to whether we should require/not require key IDs anyway, would love to hear some thoughts on this

Reply:
I'm happy to require kids because:
1. You're going to need them if you want to rotate keys and maintain your sanity
2. Our tool keygen requires them

So if we enforce that the key always has an ID for both signing and verification, users who use the 'blessed path' aren't going to fail.

One thing we could do is generate a kid if user's can't be bothered to supply one. I should have thought of that when reviewing this: https://github.com/buildkite/agent/pull/2415

If we do that then user's don't have think about it

Reply:
👀 https://github.com/buildkite/agent/pull/2422